src/main/java/frc/robot/subsystems/RollerSubsystem.java

```diff
@@ -187,23 +203,31 @@ private HeldPiece getColorSensorPiece() {
         double blue = detectedColor.blue * 255;
 
         // for tuning
-        // entry.setValue(red);
-        // gentry.setValue(green);
-        // bentry.setValue(blue);
+        entry.setValue(red);
+        gentry.setValue(green);
+        bentry.setValue(blue);
 
+        double emptyDist = Math.pow(emptyRed - red, 2) + Math.pow(emptyGreen - green, 2) + Math.pow(emptyBlue - blue, 2);
+        double coneDist = Math.pow(coneRed - red, 2) + Math.pow(coneGreen - green, 2) + Math.pow(coneBlue - blue, 2);
+        double cubeDist = Math.pow(cubeRed - red, 2) + Math.pow(cubeGreen - green, 2) + Math.pow(cubeBlue - blue, 2);
+        double minDist = Math.min(emptyDist, Math.min(coneDist, cubeDist));
+
+        System.out.println("empty -  " + emptyDist + " cone - " + coneDist + " cube - " + cubeDist);
         // TODO: modify `heldPiece` based on color sensor data
-        if (red > cubeRedMin && red < cubeRedMax && blue > cubeBlueMin && blue < cubeBlueMax && green < cubeGreenMax && green > cubeGreenMin){
-            // System.out.println("cube");
+        if (minDist == cubeDist){
```

Comment:
Also, doesn't using `==` instead of some kind of threshold make this require extremely precise readings? We probably won't get this consistency, especially when lighting changes between rooms / comps / etc.

Reply:
re-implemented for less comparisons